### PR TITLE
ext/hal/microchip/mec/mec1501 Header update Port80 capture, timer, eS…

### PR DESCRIPTION
**Pull request by scottwcpg**
…PI, TFDP

Updated MEC1501 HAL headers. Add interrupt routing defines for timers. Fix
eSPI virtual wire source access. Add new headers for Port80 capture and
TFDP.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

### CODE REVIEW

**Review by franciscomunoz — CHANGES_REQUESTED** [review]
@scottwcpg  You commit messages must be below max length(72  characters) per line.

**Comment by scottwcpg** [comment]
> @scottwcpg You commit messages must be below max length(72 characters) per line.

Updated commit message to reduce line length and pushed change.

File: ext/hal/microchip/mec/mec1501/component/espi_vw.h [hunk 1]
```diff
@@ -420,7 +420,7 @@ inline void
 mec_espi_msvw_irq_sel_set(ESPI_MSVW_REG * p, enum espi_msvw_src src,
 			  enum espi_msvw_irq_sel isel)
 {
-	volatile uint8_t *psrc = (volatile uint8_t *)p->SRC_IRQ_SEL;
+	volatile uint8_t *psrc = (volatile uint8_t *)&p->SRC_IRQ_SEL;
```
Comment (albertofloyd) [inline ext/hal/microchip/mec/mec1501/component/espi_vw.h]: Simplify these 2 lines to 1
p->SRC_IRQ_SEL = (isel << src);

Comment (scottwcpg) [reply ext/hal/microchip/mec/mec1501/component/espi_vw.h]: The select fields are on byte boundaries and enum espi_msvw_src is one of [0, 1, 2, 3].  Instead:
p->SRC_IRQ_SEL = (isel << (src << 3))
which produces shifts to each byte.

Comment (scottwcpg) [reply ext/hal/microchip/mec/mec1501/component/espi_vw.h]: Also using a 32-bit write would overwrite the other 3 VWire's IRQ select fields in this 32-bit word. This is one reason I used byte access. The fields are located on byte boundaries so one can use byte access to eliminate read-modify-write. I will be adding comments to the inline functions using this technique.

**Comment by scottwcpg** [comment]
Hi Alberto,

I pushed a rebase of espi_vw.h with comments on the inline functions using casting to get byte access.

Scott

**Comment by albertofloyd** [comment]
@andrewboie  could you review this?

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by albertofloyd** [comment]
@tbursztyka any comments on this?

**Review by franciscomunoz — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**